ui/hooks/useGasFeeEstimates.test.js

```diff
@@ -0,0 +1,238 @@
+import { cleanup, renderHook } from '@testing-library/react-hooks';
+import { useSelector } from 'react-redux';
+import { GAS_ESTIMATE_TYPES } from '../../shared/constants/gas';
+import createRandomId from '../../shared/modules/random-id';
+import {
+  getGasEstimateType,
+  getGasFeeEstimates,
+  isEIP1559Network,
+} from '../ducks/metamask/metamask';
+import {
+  disconnectGasFeeEstimatePoller,
+  getGasFeeEstimatesAndStartPolling,
+} from '../store/actions';
+import { useGasFeeEstimates } from './useGasFeeEstimates';
+
+jest.mock('../store/actions', () => ({
+  disconnectGasFeeEstimatePoller: jest.fn(),
+  getGasFeeEstimatesAndStartPolling: jest.fn(),
+}));
+
+jest.mock('react-redux', () => {
+  const actual = jest.requireActual('react-redux');
+
+  return {
+    ...actual,
+    useSelector: jest.fn(),
+  };
+});
+
+const DEFAULT_OPTS = {
+  isEIP1559Network: false,
+  gasEstimateType: GAS_ESTIMATE_TYPES.LEGACY,
+  gasFeeEstimates: {
+    low: '10',
+    medium: '20',
+    high: '30',
+  },
+};
+
+const generateUseSelectorRouter = (opts = DEFAULT_OPTS) => (selector) => {
+  if (selector === isEIP1559Network) {
+    return opts.isEIP1559Network ?? DEFAULT_OPTS.isEIP1559Network;
+  }
+  if (selector === getGasEstimateType) {
+    return opts.gasEstimateType ?? DEFAULT_OPTS.gasEstimateType;
+  }
+  if (selector === getGasFeeEstimates) {
+    return opts.gasFeeEstimates ?? DEFAULT_OPTS.gasFeeEstimates;
+  }
+  return undefined;
+};
+
+describe('useGasFeeEstimates', () => {
+  let tokens = [];
+  beforeEach(() => {
+    jest.clearAllMocks();
+    tokens = [];
+    getGasFeeEstimatesAndStartPolling.mockImplementation(() => {
+      const token = createRandomId();
+      tokens.push(token);
+      return Promise.resolve(token);
+    });
+    disconnectGasFeeEstimatePoller.mockImplementation((token) => {
+      tokens = tokens.filter((tkn) => tkn !== token);
+    });
+    useSelector.mockImplementation(generateUseSelectorRouter());
+  });
+
+  it('registers with the controller', () => {
+    renderHook(() => useGasFeeEstimates());
+    expect(tokens).toHaveLength(1);
+  });
+
+  it('clears token with the controller on unmount', async () => {
+    renderHook(() => useGasFeeEstimates());
+    expect(tokens).toHaveLength(1);
+    const expectedToken = tokens[0];
+    await cleanup();
+    expect(getGasFeeEstimatesAndStartPolling).toHaveBeenCalledTimes(1);
+    expect(disconnectGasFeeEstimatePoller).toHaveBeenCalledWith(expectedToken);
+    expect(tokens).toHaveLength(0);
+  });
+
+  it('works with LEGACY gas prices', () => {
+    const {
+      result: { current },
+    } = renderHook(() => useGasFeeEstimates());
+    expect(current).toMatchObject({
+      gasFeeEstimates: DEFAULT_OPTS.gasFeeEstimates,
+      gasEstimateType: GAS_ESTIMATE_TYPES.LEGACY,
+      estimatedGasFeeTimeBounds: undefined,
+      isGasEstimatesLoading: false,
+    });
+  });
+
+  it('works with ETH_GASPRICE gas prices', () => {
```

Comment:
Just a tiny thing, but since "gas price" are 2 words, don't we want to call it: `ETH_GAS_PRICE`?

Reply:
THis is the capitalized version of the method eth_gasPrice